src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for converting between `std::array` and boost
+/// quaternions.
+
+#pragma once
+
+#include <array>
+#include <boost/math/quaternion.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+/// \brief Convert a `boost::math::quaternion` to a `std::array<DataVector, 1>`
+///
+/// \details Functions of time return `std::array<DataVector, N>`s from their
+/// `func` members where `N` is the number of derivatives of the function.
+/// For quaternions, only the quaternion itself is stored so only
+/// need to create a `std::array<DataVector, 1>` to return
+template <typename T>
+std::array<DataVector, 1> quaternion_to_array_dv(
```

Comment:
I think these functions should drop the one element arrays and just manipulate the DataVectors.  Putting the result into an array at the call site is easy and it makes these functions more flexible.